lib/inc/internal/facts/resolvers/fips_resolver.hpp

```diff
@@ -0,0 +1,46 @@
+/**
+* @file
+* Declares the base fips fact resolver.
+*/
+#pragma once
+
+#include <facter/facts/resolver.hpp>
+
+namespace facter { namespace facts { namespace resolvers {
+
+
+    /**
+     * Responsible for resolving fips facts.
+     */
+    struct fips_resolver : resolver
+    {
+        fips_resolver();
+
+        /**
+         *  Represents fips data.
+         */
+        struct data
+        {
+            /**
+             * Stores the is_fips_mode_enabled data.
+             */
+            bool is_fips_mode_enabled;
+        };
+
+        /**
+         * Called to resolve all facts the resolver is responsible for.
+         * @param facts The fact collection that is resolving facts.
+         */
+        virtual void resolve(collection& facts) override;
+
+        /**
+ 	 * REMIND: Do we need this???
```

Comment:
Yes, this should probably stay here

Reply:
Ah, and this method and the data struct should probably also be under a `protected:` line - most other resolvers tend to set the constructor and `resolve` method to public and default to protected for other members.

Reply:
k..